### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by wuder4**
## Changes
* Making minor changes to automated test cases
* Adding bug folder with few example test cases

### CODE REVIEW

File: tests/Bugs/anonapicat.robot [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	Click Element	id=frontpage-button
+	confirm page loaded	Users
+	Click Element	id=apis-button
+	confirm page loaded	Sort by
+
+Apis sorting
+	Select From List By Value	id=sort-select	bookmarkCount
+	Location Should Contain	By=bookmark
+	Click Element	id=sortDirection-descending
+	Location Should Contain	Direction=descending
```
Comment (Nazarah) [inline tests/Bugs/anonapicat.robot]: Change the ID for ascending sorting

File: tests/Bugs/anonapicat.robot [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	Select From List By Value	id=sort-select	created_at
+	Location Should Contain	By=created
+	Click Element	id=sortDirection-ascending
+	Location Should Contain	Direction=ascending
```
Comment (Nazarah) [inline tests/Bugs/anonapicat.robot]: change the id for ascending sorting.

File: tests/Bugs/anonapicat.robot [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	Select From List By Value	id=sort-select	averageRating
+	Location Should Contain	By=average
+	Select From List By Value	id=sort-select	name
+	Location Should Contain	By=name
```
Comment (Nazarah) [inline tests/Bugs/anonapicat.robot]: change it with the current values in dropdown: Names (A-Z) and Names (Z-A)

File: tests/Gate_1/4.apicat.robot [hunk 4]
```diff
@@ -79,4 +77,4 @@ confirm page loaded
 	Wait Until Page Contains 	${searchkey}
 
 Logout of apinf
-	Click Element 	id=footer-signout
\ No newline at end of file
+	Click Element 	id=footer-signout
```
Comment (Nazarah) [inline tests/Gate_1/4.apicat.robot]: add the line "Focus		id=footer-signout" before this line

File: tests/Gate_1/5.proxy.robot [hunk 5]
```diff
@@ -72,4 +74,4 @@ confirm page loaded
 	Wait Until Page Contains 	${searchkey}
 
 Logout of apinf
-	Click Element 	id=footer-signout
\ No newline at end of file
+	Click Element 	id=footer-signout
```
Comment (Nazarah) [inline tests/Gate_1/5.proxy.robot]: add the line "Focus		id=footer-signout" before this line

File: tests/Bugs/underscorereg.robot [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	confirm page loaded	Users
+	Click Element	id=signup-button
+	confirm page loaded	Username
+
+confirm page loaded
+	[Arguments] 	${searchkey}
+	Wait Until Page Contains 	${searchkey}
+
+Logout of apinf
+	Click Element 	id=footer-signout
```
Comment (Nazarah) [inline tests/Bugs/underscorereg.robot]: add the line "Focus		id=footer-signout" before this line

**Comment by Nazarah** [comment]
Please add option to test selecting lifecycle status while adding an API.
As discussed before, the code could be the following:

**Function to add an API**

Create API
	[Arguments] 	${name}	${description}	${url} ${Lifecycle}
	Input Text 	id=api-name	${name}
	Input Text	id=api-description	${description}
	Input Text	id=api-url	${url}
	Select From List By Value	id=api-lifecyclestatus	${Lifecycle}
	Sleep	0.3
	Click Element	id=submitapi-button

**Testcase**

Create new api (positive)
	Go to addapi
	Create API	Kissa	${EMPTY}	https://www.google.com design
	confirm page loaded	Kissa

In for the negative testcases to add an API, you can leave the argument for lifecycle empty.

**Comment by marla-singer** [comment]
@wuder4 @Nazarah What status of this PR? Should it be improved?

**Comment by Nazarah** [comment]
@marla-singer I'd verify the change after I finish up manual testing on staging 0.52 release

**Comment by marla-singer** [comment]
@preriasusi What to do with this PR?

**Comment by marla-singer** [comment]
@Nazarah What to do with this PR?

**Comment by preriasusi** [comment]
ping @Nazarah

**Comment by Nazarah** [comment]
@marla-singer I pulled the latest update from the branch in my local host and saw that the changes requested were not made.
However, I added the 1st change request (e.g. adding "Focus id= footer-signout" under Logout of apinf section in underscorereg.robot file)
After saving the file i ran meteor and started the application in my localhost.
Then i opened up the python command line for desktop and tried to executed the following lines: 
pybot underscorered.robot
But the editor gave me syntax error.
image

AM not sure what I am doing here wrong to be honest.
Can you please help me up?
For this reason I didn't push and updates in my code in this branch

**Comment by marla-singer** [comment]
@Nazarah I ran test on local using `robot` command
1. Make sure`robot` is installed on your PC
2. Go to folder with tets `path/to/platform/test/Bugs`
3. Open the terminal and run `robot underscorereg.robot`

About `pybot` I have no idea

**Comment by marla-singer** [comment]
@Nazarah Also about your comment by 4 December - please create a new issue about that with label "Automate testing"
I gonna do rebase these commits, add an ID attribute for Site logo (otherwise a test fails) and let's merge it

File: tests/Bugs/underscorereg.robot (hunk 6 above)
Comment (marla-singer) [reply tests/Bugs/underscorereg.robot]: The test works correct w/out suggested line. Why does it need?

Comment (marla-singer) [reply tests/Bugs/underscorereg.robot]: Sorry. I saw later that this requested is done

**Pull request merged**